### DISCUSSION TITLE
Failed to build CppSharp.dll

### DISCUSSION THREAD

**Discussion by ZCH2005** [mono/CppSharp#1847] [Q&A]
I start VS2022 developer command prompt as administrator, and issue command **"C:\Program Files\Git\bin\sh.exe" build.sh -configuration Release -platform x64 -target-framework net8.0** to build. Then I got following link errors:

Parser.obj : error LNK2019: unresolved external symbol LLVMInitializeAArch64TargetInfo referenced in function "public:
void __cdecl CppSharp::CppParser::Parser::Setup(bool)" (?Setup@Parser@CppParser@CppSharp@@QEAAX_N@Z) [C:\Users\CZhang6\
Downloads\CppSharp-1.1.2\CppSharp-1.1.2\build\vs2022\projects\CppSharp.CppParser.vcxproj]
lldCOFF.lib(Driver.obj) : error LNK2001: unresolved external symbol LLVMInitializeAArch64TargetInfo [C:\Users\CZhang6\D
ownloads\CppSharp-1.1.2\CppSharp-1.1.2\build\vs2022\projects\CppSharp.CppParser.vcxproj]
Parser.obj : error LNK2019: unresolved external symbol LLVMInitializeAArch64Target referenced in function "public: void
 __cdecl CppSharp::CppParser::Parser::Setup(bool)" (?Setup@Parser@CppParser@CppSharp@@QEAAX_N@Z) [C:\Users\CZhang6\Down
loads\CppSharp-1.1.2\CppSharp-1.1.2\build\vs2022\projects\CppSharp.CppParser.vcxproj]
lldCOFF.lib(Driver.obj) : error LNK2001: unresolved external symbol LLVMInitializeAArch64Target [C:\Users\CZhang6\Downl
oads\CppSharp-1.1.2\CppSharp-1.1.2\build\vs2022\projects\CppSharp.CppParser.vcxproj]
Parser.obj : error LNK2019: unresolved external symbol LLVMInitializeAArch64TargetMC referenced in function "public: vo
id __cdecl CppSharp::CppParser::Parser::Setup(bool)" (?Setup@Parser@CppParser@CppSharp@@QEAAX_N@Z) [C:\Users\CZhang6\Do
wnloads\CppSharp-1.1.2\CppSharp-1.1.2\build\vs2022\projects\CppSharp.CppParser.vcxproj]
lldCOFF.lib(Driver.obj) : error LNK2001: unresolved external symbol LLVMInitializeAArch64TargetMC [C:\Users\CZhang6\Dow
nloads\CppSharp-1.1.2\CppSharp-1.1.2\build\vs2022\projects\CppSharp.CppParser.vcxproj]
Parser.obj : error LNK2019: unresolved external symbol LLVMInitializeAArch64AsmPrinter referenced in function "public:
struct CppSharp::CppParser::ParserResult * __cdecl CppSharp::CppParser::Parser::Compile(class std::basic_string<char,st
ruct std::char_traits<char>,class std::allocator<char> > const &)" (?Compile@Parser@CppParser@CppSharp@@QEAAPEAUParserR
esult@23@AEBV?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@@Z) [C:\Users\CZhang6\Downloads\CppSharp-1.1.
2\CppSharp-1.1.2\build\vs2022\projects\CppSharp.CppParser.vcxproj]
lldCOFF.lib(Driver.obj) : error LNK2001: unresolved external symbol LLVMInitializeAArch64AsmPrinter [C:\Users\CZhang6\D
ownloads\CppSharp-1.1.2\CppSharp-1.1.2\build\vs2022\projects\CppSharp.CppParser.vcxproj]
Parser.obj : error LNK2019: unresolved external symbol LLVMInitializeAArch64AsmParser referenced in function "public: v
oid __cdecl CppSharp::CppParser::Parser::Setup(bool)" (?Setup@Parser@CppParser@CppSharp@@QEAAX_N@Z) [C:\Users\CZhang6\D
ownloads\CppSharp-1.1.2\CppSharp-1.1.2\build\vs2022\projects\CppSharp.CppParser.vcxproj]
lldCOFF.lib(Driver.obj) : error LNK2001: unresolved external symbol LLVMInitializeAArch64AsmParser [C:\Users\CZhang6\Do
wnloads\CppSharp-1.1.2\CppSharp-1.1.2\build\vs2022\projects\CppSharp.CppParser.vcxproj]
..\..\..\bin\Release_x64\CppSharp.CppParser.dll : fatal error LNK1120: 5 unresolved externals [C:\Users\CZhang6\Downloa
ds\CppSharp-1.1.2\CppSharp-1.1.2\build\vs2022\projects\CppSharp.CppParser.vcxproj]

**Comment by tritao** [comment]
Seems like it was caused by this commit: https://github.com/mono/CppSharp/commit/bbb0693ef53a36db667ee4f4e53e35b4fca87217
It is strange because its building fine on CI, it should be using this package for LLVM: https://github.com/mono/CppSharp/releases/download/CppSharp/llvm-6eb36a-windows-vs2022-x64-RelWithDebInfo.7z

But in your case its not finding the correct symbols.

**Comment by tritao** [comment]
Did you previously build CppSharp on this machine?

**Reply by ZCH2005** [reply]
This is the first time to build CppCSharp on this machine. Actually, the correct LLVM package was downloaded:

C:\Users\CZhang6\Downloads\CppSharp-1.1.2\build>"C:\Program Files\Git\bin\sh.exe" build.sh -configuration Release -platform=X64 -target-framework net8.0
net8.0
Archive llvm-6eb36a-windows-vs2022-x64-RelWithDebInfo.7z already exists.
Using debug LLVM build: C:/Users/CZhang6/Downloads/CppSharp-1.1.2/build/llvm/llvm-6eb36a-windows-vs2022-x64-Debug
Using release LLVM build: C:/Users/CZhang6/Downloads/CppSharp-1.1.2/build/llvm/llvm-6eb36a-windows-vs2022-x64-RelWithDebInfo
Generating CppSharp build configuration file 'BuildConfig.cs'
Building configurations...
Running action 'vs2022'...
Done (216ms).

  Determining projects to restore...
  All projects are up-to-date for restore.

  CppSharp -> C:\Users\CZhang6\Downloads\CppSharp-1.1.2\bin\Release_x64\CppSharp.dll
  CppSharp.AST -> C:\Users\CZhang6\Downloads\CppSharp-1.1.2\bin\Release_x64\CppSharp.AST.dll
     Creating library ..\..\..\bin\Release_x64\CppSharp.CppParser.lib and object ..\..\..\bin\Release_x64\CppSharp.CppP
  arser.exp
Parser.obj : error LNK2019: unresolved external symbol LLVMInitializeAArch64TargetInfo referenced in function "public:
void __cdecl CppSharp::CppParser::Parser::Setup(bool)" (?Setup@Parser@CppParser@CppSharp@@QEAAX_N@Z) [C:\Users\CZhang6\
Downloads\CppSharp-1.1.2\build\vs2022\projects\CppSharp.CppParser.vcxproj]
lldCOFF.lib(Driver.obj) : error LNK2001: unresolved external symbol LLVMInitializeAArch64TargetInfo [C:\Users\CZhang6\D
ownloads\CppSharp-1.1.2\build\vs2022\projects\CppSharp.CppParser.vcxproj]
Parser.obj : error LNK2019: unresolved external symbol LLVMInitializeAArch64Target referenced in function "public: void
 __cdecl CppSharp::CppParser::Parser::Setup(bool)" (?Setup@Parser@CppParser@CppSharp@@QEAAX_N@Z) [C:\Users\CZhang6\Down
loads\CppSharp-1.1.2\build\vs2022\projects\CppSharp.CppParser.vcxproj]
lldCOFF.lib(Driver.obj) : error LNK2001: unresolved external symbol LLVMInitializeAArch64Target [C:\Users\CZhang6\Downl
oads\CppSharp-1.1.2\build\vs2022\projects\CppSharp.CppParser.vcxproj]
Parser.obj : error LNK2019: unresolved external symbol LLVMInitializeAArch64TargetMC referenced in function "public: vo
id __cdecl CppSharp::CppParser::Parser::Setup(bool)" (?Setup@Parser@CppParser@CppSharp@@QEAAX_N@Z) [C:\Users\CZhang6\Do
wnloads\CppSharp-1.1.2\build\vs2022\projects\CppSharp.CppParser.vcxproj]
lldCOFF.lib(Driver.obj) : error LNK2001: unresolved external symbol LLVMInitializeAArch64TargetMC [C:\Users\CZhang6\Dow
nloads\CppSharp-1.1.2\build\vs2022\projects\CppSharp.CppParser.vcxproj]
Parser.obj : error LNK2019: unresolved external symbol LLVMInitializeAArch64AsmPrinter referenced in function "public:
struct CppSharp::CppParser::ParserResult * __cdecl CppSharp::CppParser::Parser::Compile(class std::basic_string<char,st
ruct std::char_traits<char>,class std::allocator<char> > const &)" (?Compile@Parser@CppParser@CppSharp@@QEAAPEAUParserR
esult@23@AEBV?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@@Z) [C:\Users\CZhang6\Downloads\CppSharp-1.1.
2\build\vs2022\projects\CppSharp.CppParser.vcxproj]
lldCOFF.lib(Driver.obj) : error LNK2001: unresolved external symbol LLVMInitializeAArch64AsmPrinter [C:\Users\CZhang6\D
ownloads\CppSharp-1.1.2\build\vs2022\projects\CppSharp.CppParser.vcxproj]
Parser.obj : error LNK2019: unresolved external symbol LLVMInitializeAArch64AsmParser referenced in function "public: v
oid __cdecl CppSharp::CppParser::Parser::Setup(bool)" (?Setup@Parser@CppParser@CppSharp@@QEAAX_N@Z) [C:\Users\CZhang6\D
ownloads\CppSharp-1.1.2\build\vs2022\projects\CppSharp.CppParser.vcxproj]
lldCOFF.lib(Driver.obj) : error LNK2001: unresolved external symbol LLVMInitializeAArch64AsmParser [C:\Users\CZhang6\Do
wnloads\CppSharp-1.1.2\build\vs2022\projects\CppSharp.CppParser.vcxproj]
..\..\..\bin\Release_x64\CppSharp.CppParser.dll : fatal error LNK1120: 5 unresolved externals [C:\Users\CZhang6\Downloa
ds\CppSharp-1.1.2\build\vs2022\projects\CppSharp.CppParser.vcxproj]
